test/wrapeth/provider_test.exs

```diff
@@ -0,0 +1,109 @@
+defmodule Wrapeth.ProviderTest do
+  use ExUnit.Case, async: true
+  import Mox
+
+  defmodule TestProvider do
+    use Wrapeth.Provider, otp_app: :wrapeth
+  end
+
+  # Make sure mocks are verified when the test exits
+  setup :verify_on_exit!
+
+  @valid_address "0x4Bd040d48fdD1C667fcE5fdDd326681766E6ad91"
+  @invalid_addres "0y4Bd040d48fdD1C667fcE5fdDd326681766E6ad91"
+  @invalid_block_number "0y9710d"
```

Comment:
Do you prefer module attributes in tests over setup? Why?

Reply:
To be honest, I initially went with the first approach that came to my mind. After looking into the setup functions, I found it to be a more fine-grained approach. I have decided to switch to using the setup function in the last commit.